src/shims/unix/socket.rs

```diff
@@ -42,17 +105,40 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
     ) -> InterpResult<'tcx, Scalar<Provenance>> {
         let this = self.eval_context_mut();
 
-        let _domain = this.read_scalar(domain)?.to_i32()?;
-        let _type_ = this.read_scalar(type_)?.to_i32()?;
-        let _protocol = this.read_scalar(protocol)?.to_i32()?;
+        let domain = this.read_scalar(domain)?.to_i32()?;
+        let type_ = this.read_scalar(type_)?.to_i32()?;
+        let protocol = this.read_scalar(protocol)?.to_i32()?;
         let sv = this.deref_pointer(sv)?;
 
-        // FIXME: fail on unsupported inputs
+        // Fail on unsupported input
+        if domain != this.eval_libc_i32("AF_UNIX") && domain != this.eval_libc_i32("AF_LOCAL") {
+            throw_unsup_format!("socketpair: Only AF_UNIX and AF_LOCAL are supported");
+        } else if type_ != this.eval_libc_i32("SOCK_STREAM")
+            && type_ != this.eval_libc_i32("SOCK_DGRAM")
+        {
+            throw_unsup_format!("socketpair: Only SOCK_STREAM and SOCK_DGRAM are suppported");
+        } else if protocol != 0 {
+            throw_unsup_format!("socketpair: Unsupported socket protocol is used");
+        }
+
+        let buf1 = Rc::new(RefCell::new(VecDeque::with_capacity(3)));
```

Comment:
Why `3`?

Reply:
Oops, that's a typo. I intended to put ``with_capacity(212992)``, which is the magic number that is going to be a named constant soon.

Reply:
Should we really consume 400k RAM for each `socketpair`?
Does the kernel pre-allocate the entire buffer or grow it up to the given size as needed?

Reply:
We can safely just use an empty VecDeque. It will grow as it needs more size, and we have a hard limit on the 212992 due to the separate check for the length.